elyra/pipeline/processor_kfp.py

```diff
@@ -68,10 +69,14 @@ def process(self, pipeline):
 
             self.log.debug("Creating temp directory %s", temp_dir)
 
+            engine = runtime_configuration.metadata.get('engine')
             # Compile the new pipeline
             try:
                 pipeline_function = lambda: self._cc_pipeline(pipeline, pipeline_name)  # nopep8 E731
-                kfp.compiler.Compiler().compile(pipeline_function, pipeline_path)
+                if ('Argo' == engine):
```

Comment:
In a way this is perhaps not consistent with schema's default, which is `Argo`. If, for whatever reason, the runtime configuration does contain an invalid value `XYZ`, the  logic would run the Tekton compiler.

Reply:
This is also inconsistent with the logic in https://github.com/elyra-ai/elyra/blob/bde3b4f3d284407dc0f42a3c80f8983e1066b749/elyra/pipeline/processor_kfp.py#L157, where we test for `Tekton`.

Reply:
other values should not really come in as the metadata is validated and invalid ones will not be shown in the user interface....